node/src/components/storage.rs

```diff
@@ -543,9 +542,7 @@ impl Storage {
                 // Remove blocks that are in to-be-upgraded eras, but have obsolete protocol
                 // versions - they were most likely created before the upgrade and should be
                 // reverted.
-                if block_header.era_id() >= invalid_era
-                    && block_header.protocol_version() < protocol_version
-                {
+                if block_header.era_id() >= invalid_era {
```

Comment:
But what if there was an emergency upgrade in the past and we've already correctly handled it and continued. If we then restart the node won't this remove a lot of perfectly valid blocks?